src/Juvix/Core.hs

```diff
@@ -1,12 +1,126 @@
 module Juvix.Core
-  ( module Juvix.Core.MainLang,
+  ( hrToIR,
+    irToHR,
     module Juvix.Core.Erasure,
-    module Juvix.Core.SemiRing,
-    module Juvix.Core.Parser,
+    module Juvix.Core.Usage,
   )
 where
 
 import Juvix.Core.Erasure
-import Juvix.Core.MainLang
-import Juvix.Core.Parser
-import Juvix.Core.SemiRing
+import qualified Juvix.Core.HR as HR
+import qualified Juvix.Core.IR as IR
+import Juvix.Core.Usage
+import Juvix.Core.Utility
+import Juvix.Library
+
+-- contract: no shadowing
+-- TODO - handle this automatically by renaming shadowed vars
+hrToIR ∷ HR.Term primTy primVal → IR.Term primTy primVal
```

Comment:
Hmm these changes should probably be added to a `Core/Translate.hs` or something like that

Reply:
Agreed - moved them there.